bin/dfx-canister-url

```diff
@@ -63,6 +64,7 @@ URL_TYPES=(static api)
 clap.define short=n long=network desc="The dfx network to use" variable=DFX_NETWORK default="local"
 clap.define short=c long=canister desc="The canister name" variable=DFX_CANISTER
 clap.define short=t long=type desc="The canister type (options: ${URL_TYPES[*]})" variable=DFX_URL_TYPE default=static
+clap.define short=r long=raw desc="Provide a raw URL" variable=DFX_RAW nargs=0 default=""
```

Comment:
Note:  It _may_ be better to have raw as a third type of URL.  Semantically, which makes more sense?  I'm not sure.

Reply:
Is it possible to have a raw API URL that's different from the corresponding raw static URL or does that not make any sense?

I think the answer to this question determines whether it's a separate type of not.

Reply:
Discussed - raw API URLs are not a thing.  I have added a test "There are no raw API URLs" and updated the code accordingly.